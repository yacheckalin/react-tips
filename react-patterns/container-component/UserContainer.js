import React from "react";
import axios from "axios";

const UsersList = ({ data }) => (
  <ul className="collection">
    {data.map((user) => (
      <li key={user.id} className="collection-item">
        <span>{user.id}</span> {user.name}
        <span className="secondary-content">{user.email}</span>
      </li>
    ))}
  </ul>
);

class UserContainer extends React.Component {
  constructor() {
    super();
    this.state = { data: [] };
  }
  async componentDidMount() {
    const { data } = await axios.get(
      "https://jsonplaceholder.typicode.com/users"
    );
    this.setState({ data });
  }
  render() {
    return <UsersList data={this.state.data} />;
  }
}

export default UserContainer;
