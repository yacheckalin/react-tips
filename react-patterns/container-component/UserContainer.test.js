import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import UserContainer from "./UserContainer";
import axios from "axios";

jest.mock("axios");

describe("<UserContainer />", () => {
  it("should fetch user data", async () => {
    const data = {};

    axios.get.mockImplementationOnce(() => Promise.resolve(data));
  });

  it("renders and displays properly", () => {
    const wrapper = shallow(<UserContainer />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
