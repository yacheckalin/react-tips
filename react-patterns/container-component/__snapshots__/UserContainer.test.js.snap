// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserContainer /> renders and displays properly 1`] = `
<UsersList
  data={Array []}
/>
`;
